views/home.ejs

```diff
@@ -22,4 +23,18 @@
         <% } %>
     <% } %>
 </div>
+<script>
+    function favorite(postId) {
+        var userId = document.getElementById('currentUserId').value;
+        $.post(`post/${postId}/favorite?format=json`, (post) => {
+            $(`#post-favorite-count-${postId}`).html(post.favorites.length);
+
+            if(post.favorites.includes(userId)){
+                $(`#post-favorite-button-${postId}`).attr('fill', '#EAE34B')
+            } else {
+                $(`#post-favorite-button-${postId}`).attr('fill', 'none')
+            }
+        });
+    }
+</script>
```

Comment:
Wenn du die favorites methode so oft schreiben musst, kannst du sie dann nicht in eine .js Datei schreiben (siehe js directory) und dann im script tag einfach darauf verlinken? Spart codezeilen und alle calls nutzen die selbe Methode.